### DISCUSSION TITLE
FeTS tool return "qt.qpa.xcb: could not connect to display" error when launched

### DISCUSSION THREAD

**Discussion by marufmore** [FeTS-AI/Front-End#55] [Q&A] [answered]
## Expected Behavior
The FeTS software is supposed to start

## Actual Behavior
It returns the following error message
qt.qpa.xcb: could not connect to display
qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "" even though it was found.
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: xcb.

Aborted
## Steps to Reproduce the Problem

  1. Tried searching for solutions on stackoverflow.
  2.
  3.

## Specifications

  - Version: FeTS 0.0.7
  - Platform: Ubuntu 18.04
  - Subsystem: WSL

**Comment by sarthakpati** [comment, marked as answer]
Hi @marufmore,

This issue is because a display was not detected on your machine. I see that you are running WSL. Can you specify which version of WSL you are running by running the following command:
```bash
wsl --list --verbose
```

WSL-2 has a built-in way of initializing X11 displays. See [this answer on SO](https://stackoverflow.com/a/68898469/1228757) for more details. For WSL-1, there are a few steps to get this done [[ref](https://medium.com/javarevisited/using-wsl-2-with-x-server-linux-on-windows-a372263533c3)]:
- Install VcXsrv: https://sourceforge.net/projects/vcxsrv/
- Create a new desktop shortcut and use the following command. Add the following command in the properties → shortcut →target:
		`"C:\Program Files\VcXsrv\vcxsrv.exe" :0 -ac -terminate -lesspointer -multiwindow -clipboard -wgl -dpi auto`
 - Verify that it is running by executing the command "netstat -abno|findstr 6000" in a powershell terminal with admin rights
 - Add the following to your "~/.bashrc" file in WSL2 account: 
		`alias x_display="DISPLAY=$(cat /etc/resolv.conf | grep nameserver | awk '{print $2}'):0"`
 - Now, you should be able to run "x_display xclock" or any other X11 application.

**Comment by marufmore** [comment]
I'm running WSL2. It works now.
